OpenRA.Mods.Common/Widgets/Logic/Installation/InstallFromSourceLogic.cs

```diff
@@ -407,18 +441,31 @@ void ShowBackRetry(Action retryAction)
 			primaryButton.Text = modData.Translation.GetString(Retry);
 			primaryButton.Visible = true;
 
-			secondaryButton.OnClick = Ui.CloseWindow;
-			secondaryButton.Text = modData.Translation.GetString(Back);
 			secondaryButton.Visible = true;
-			secondaryButton.Disabled = false;
+			secondaryButton.OnClick = TryLocate;
```

Comment:
a. use a better name for buttons? `selectDirectoryButton`?

b. best to disable `back` and `retry` buttons while resolve task is is in progress. file operations may take a while.

Reply:
I agree, the button naming isn't the best here